homeassistant/components/schedule/__init__.py

```diff
@@ -0,0 +1,305 @@
+        [{CONF_ID: id_, **cfg} for id_, cfg in config.get(DOMAIN, {}).items()]
+    )
+    await storage_collection.async_load()
+
+    StorageCollectionWebsocket(
+        storage_collection,
+        DOMAIN,
+        DOMAIN,
+        BASE_SCHEMA | STORAGE_SCHEDULE_SCHEMA,
+        BASE_SCHEMA | STORAGE_SCHEDULE_SCHEMA,
+    ).async_setup(hass)
+
+    async def reload_service_handler(service_call: ServiceCall) -> None:
+        """Reload yaml entities."""
+        conf = await component.async_prepare_reload(skip_reset=True)
+        if conf is None:
+            conf = {DOMAIN: {}}
+        await yaml_collection.async_load(
+            [{CONF_ID: id_, **cfg} for id_, cfg in conf.get(DOMAIN, {}).items()]
+        )
+
+    async_register_admin_service(
+        hass,
+        DOMAIN,
+        SERVICE_RELOAD,
+        reload_service_handler,
+    )
+
+    return True
+
+
+class ScheduleStorageCollection(StorageCollection):
+    """Schedules stored in storage."""
+
+    SCHEMA = vol.Schema(BASE_SCHEMA | STORAGE_SCHEDULE_SCHEMA)
+
+    async def _process_create_data(self, data: dict) -> dict:
+        """Validate the config is valid."""
+        self.SCHEMA(data)
+        return data
+
+    @callback
+    def _get_suggested_id(self, info: dict) -> str:
+        """Suggest an ID based on the config."""
+        name: str = info[CONF_NAME]
+        return name
+
+    async def _update_data(self, data: dict, update_data: dict) -> dict:
+        """Return a new updated data object."""
+        self.SCHEMA(update_data)
+        return data | update_data
+
+    async def _async_load_data(self) -> dict | None:
+        """Load the data."""
+        if data := await super()._async_load_data():
+            data["items"] = [STORAGE_SCHEMA(item) for item in data["items"]]
+        return data
+
+
+class Schedule(Entity):
+    """Schedule entity."""
+
+    _attr_has_entity_name = True
+    _attr_should_poll = False
+    _attr_state: Literal["on", "off"]
+    _config: ConfigType
+    _next: datetime
+    _unsub_update: Callable[[], None] | None = None
+    editable = True
+
+    def __init__(self, config: ConfigType) -> None:
+        """Initialize a schedule."""
+        self._config = STORAGE_SCHEMA(config)
+        self._attr_unique_id = self._config[CONF_ID]
+        self._attr_icon = self._config.get(CONF_ICON)
+        self._attr_name = self._config[CONF_NAME]
+
+    @classmethod
+    def from_yaml(cls, config: ConfigType) -> Schedule:
+        """Return entity instance initialized from yaml storage."""
+        schedule = cls(config)
+        schedule.entity_id = f"{DOMAIN}.{config[CONF_ID]}"
+        schedule.editable = False
+        return schedule
+
+    async def async_update_config(self, config: ConfigType) -> None:
+        """Handle when the config is updated."""
+        self._config = STORAGE_SCHEMA(config)
+        self._attr_icon = config.get(CONF_ICON)
+        self._attr_name = config[CONF_NAME]
+        self._update()
```

Comment:
This will schedule a new point in time listener without cleaning up any existing one. 
```suggestion
        if self._unsub_update:
            self._unsub_update()
            self._unsub_update = None

        self._update()
```

Reply:
Same, but smaller 😉 

```suggestion
        self._clean_up_listener()
        self._update()
```